src/auth.ts

```diff
@@ -215,6 +234,52 @@ export class BasicAuth extends AuthType implements IBasicAuthType {
     return deferred.promise;
   }
 
+  requestPasswordReset(data: ForgotPasswordDetails): Promise<string> {
+    var deferred = new DeferredPromise<string, Error>();
+
+    if (!data.email) {
+      deferred.reject(new Error('Email is required for password reset request.'))
+    } else {
+      this.client.post('/users/password/reset')
+          .send({
+            'app_id': this.config.get('app_id'),
+            'email': data.email,
+            'cloud_client': true
+          })
+          .end((err, res) => {
+            if (err) {
+              deferred.reject(err);
+            } else {
+              deferred.resolve(res.body);
+            }
+          });
+    }
+    return deferred.promise;
+  }
+
+  confirmPasswordReset(data: ResetPasswordDetails): Promise<string> {
+    var deferred = new DeferredPromise<string, Error>();
+
+    if (!data.code || !data.newPassword || !data.newPasswordVerify) {
+        deferred.reject(new Error('Code, new password, and new password verification are required.'));
+    } else {
+      this.client.post('/users/password')
+          .send({
+            'reset_token': data.code,
+            'new_password': data.newPassword,
+            'new_password_verify': data.newPasswordVerify
```

Comment:
The API should not need to accept a double password verification, that can happen client side. Instead this should send the email that was sent with the passwordRequest similar to my other comment.